arangod/GeneralServer/CommTask.cpp

```diff
@@ -375,6 +375,16 @@ void CommTask::finishExecution(GeneralResponse& res,
     // use "IfNotSet" to not overwrite an existing response header
     res.setHeaderNCIfNotSet(StaticStrings::XContentTypeOptions,
                             StaticStrings::NoSniff);
+
```

Comment:
do we care in case of VST? so, could it only be added for HTTP[1.1/2]?

Reply:
```
curl --http2 -v -i http://localhost:8529/
*   Trying 127.0.0.1:8529...
* Connected to localhost (127.0.0.1) port 8529 (#0)
> GET / HTTP/1.1
> Host: localhost:8529
> User-Agent: curl/7.84.0
> Accept: */*
> Connection: Upgrade, HTTP2-Settings
> Upgrade: h2c
> HTTP2-Settings: AAMAAABkAAQCAAAAAAIAAAAA
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 101 Switching Protocols
HTTP/1.1 101 Switching Protocols
< Connection: Upgrade
Connection: Upgrade
< Upgrade: h2c
Upgrade: h2c
* Received 101
* Using HTTP2, server supports multiplexing
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0

* Connection state changed (MAX_CONCURRENT_STREAMS == 32)!
* HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1)
* Connection #0 to host localhost left intact
curl: (92) HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1)
```

It seems HTTP/2 support is not fully functional. Will have to wait until it is, before testing headers for this protocol.

Reply:
Correction: The HTTP/2 error was being caused by camel-cased headers. This is now fixed with all headers being sent in lowercase. To answer the original question, `Endpoint::TransportType::HTTP` is the type for HTTP/2 requests also.